sympy/combinatorics/coset_table.py

```diff
@@ -356,10 +367,7 @@ def scan(self, alpha, word, y=None, fill=False, modified=False):
                     self.P[f][self.A_dict[word[i]]] = f_p**-1*b_p
                     self.P[b][self.A_dict_inv[word[i]]] = b_p**-1*f_p
```

Comment:
There is a `return` statement here in the Handbook. Could you check if the stops the infinite loop without `itr`?

Reply:
I tried adding the return statement. It's still the same. The problem is when `self.scan(beta, w)` is called from `look_ahead` method which is called from `process_deductions`. This happens with `coset_enumeration_c`.